src/main/kotlin/com/cultureamp/eventsourcing/PGSessionAdvisoryBookmarkLock.kt

```diff
@@ -0,0 +1,70 @@
+package com.cultureamp.eventsourcing
+
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import java.sql.SQLException
+import java.util.UUID
+
+private val logger = LoggerFactory.getLogger(PGSessionAdvisoryBookmarkLock::class.java)
+class PGSessionAdvisoryBookmarkLock(
+    private val lockTimeoutMs: Int = 60_000,
+    private val connectionProvider: () -> Connection,
+) {
+    private val lock = Any()
+    private var connection = getConnection()
+
+    fun lock(bookmarkName: String) {
+        synchronized(lock) {
+            try {
+                refreshLock(connection, bookmarkName)
+            } catch (e: Exception) {
```

Comment:
Do we want to get more specific about the scenario we're catching? The connection being closed

Reply:
Just wanted to check if we can get more specific here - if not, then it's fine